openfeature-provider/go/demo/README.md

```diff
@@ -0,0 +1,286 @@
+stringResult := client.StringValueDetails(ctx, "theme-config.primaryColor", "#000000", evalCtx)
+```
+
+### 7. Targeting Key
+```go
+evalCtx := openfeature.NewTargetingKey("session-xyz")
+boolResult := client.BooleanValueDetails(ctx, "beta-feature", false, evalCtx)
+```
+
+## Evaluation Context
+
+The demo shows different ways to create evaluation contexts:
+
+```go
+// With targeting key and attributes
+evalCtx := openfeature.NewEvaluationContext(
+    "user-123",
+    map[string]interface{}{
+        "email":  "user@example.com",
+        "region": "US",
+        "plan":   "premium",
+    },
+)
+
+// Just targeting key
+evalCtx := openfeature.NewTargetingKey("session-xyz")
+```
+
+## Troubleshooting
+
+### "FLAG_NOT_FOUND" Errors
+
+If you see flag not found errors:
+1. Verify you have flags created in your Confidence account
+2. Check that the flag names match exactly
+3. Ensure your client secret is correct
+
+### Authentication Errors
+
+If you see authentication errors:
+1. Verify your API credentials are correct
+2. Check that you're using the correct region (EU vs US)
+3. Ensure your credentials have the necessary permissions
+
+### WASM File Not Found
+
+If you see WASM file errors:
+1. Check the file path: `CONFIDENCE_WASM_FILE_PATH`
+2. Build the WASM module if needed
+3. Verify the file exists and is readable
+
+### Connection Errors
+
+If you see connection errors:
+1. Check your internet connection
+2. Verify the service addresses are correct for your region
+3. Check if you're behind a proxy or firewall
+
+## Next Steps
+
+After running the demo:
+
+1. **Create your own flags** in the Confidence dashboard
+2. **Modify the demo** to evaluate your flags
+3. **Integrate into your application** using the same pattern
+4. **Add your own evaluation context** attributes based on your use case
+
+## Integration Example
+
+Here's a minimal example of integrating the provider into your application:
+
+```go
+package main
+
+import (
+    "context"
+    "log"
+    "os"
+
+    "github.com/open-feature/go-sdk/openfeature"
+    provider "github.com/spotify/confidence-resolver-rust/openfeature-provider/go"
+    "github.com/tetratelabs/wazero"
+)
+
+func main() {
+    ctx := context.Background()
+
+    // Load WASM
+    wasmBytes, _ := os.ReadFile("resolver.wasm")
+    runtime := wazero.NewRuntime(ctx)
+    defer runtime.Close(ctx)
+
+    // Create factory
+    factory, err := provider.NewLocalResolverFactory(
+        ctx, runtime, wasmBytes,
+        "resolver.eu.confidence.dev:443",
+        "events.eu.confidence.dev:443",
+        "iam.eu.confidence.dev:443",
+        os.Getenv("CONFIDENCE_API_CLIENT_ID"),
+        os.Getenv("CONFIDENCE_API_CLIENT_SECRET"),
+    )
+    if err != nil {
+        log.Fatal(err)
+    }
+    defer factory.Shutdown(ctx)
+
+    // Register provider
+    confidenceProvider := provider.NewLocalResolverProvider(
+        factory,
+        os.Getenv("CONFIDENCE_CLIENT_SECRET"),
+    )
```

Comment:
nit: Should the integrator rather use `NewProvider` and not create the `factory` directly? Perhaps keep the factory internal if possible, to keep the integration APIs as concise and opinionated as possible